linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/bash/shared.sh

```diff
@@ -26,8 +31,13 @@ if grep -q '^GRUB_CMDLINE_LINUX=".*boot=.*"'  /etc/default/grub; then
 	sed -i 's/\(^GRUB_CMDLINE_LINUX=".*\)boot=[^[:space:]]*\(.*"\)/\1 boot=UUID='"${BOOT_UUID} \2/" /etc/default/ grub
 else
 	# no existing boot=arg is present, append it
-	sed -i 's/\(^GRUB_CMDLINE_LINUX=".*\)"/\1 boot=UUID='${BOOT_UUID}'"/'  /etc/default/grub
+	sed -i "s/\(^GRUB_CMDLINE_LINUX=\".*\)\"/\1 boot=UUID=${BOOT_UUID}\"/" /etc/default/grub
 fi
 
 # Correct the form of kernel command line for each installed kernel in the bootloader
 /sbin/grubby --update-kernel=ALL --args="fips=1 boot=UUID=${BOOT_UUID}"
+
+# Disable ed25519 key in SSH configuration (does not work in FIPS mode)
+if grep -q '^HostKey /etc/ssh/ssh_host_ed25519_key' /etc/ssh/sshd_config; then
+	sed -i 's|^HostKey /etc/ssh/ssh_host_ed25519_key|#HostKey /etc/ssh/ssh_host_ed25519_key|' /etc/ssh/sshd_config
+fi
```

Comment:
Hmmmm..... not sure that this makes sense to be in the FIPS grub rememdiation.....

Reply:
I am not sure where else you would put this. The need for this change is directly tied to enabling FIPS mode.

Reply:
This is a separate rule with separate OVAL and remediations then as the SSH HostKey's now need to be verified against FIPS.